pkg/services/ngalert/database_mig.go

```diff
@@ -0,0 +1,116 @@
+package ngalert
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addAlertDefinitionMigrations(mg *migrator.Migrator) {
+	alertDefinition := migrator.Table{
+		Name: "alert_definition",
+		Columns: []*migrator.Column{
+			{Name: "id", Type: migrator.DB_BigInt, IsPrimaryKey: true, IsAutoIncrement: true},
+			{Name: "org_id", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "name", Type: migrator.DB_NVarchar, Length: 255, Nullable: false},
```

Comment:
Don't see this being unique. Have you considered a __UID__? Will be hard to handle alert rules in multiple environments/provisioning otherwise

Reply:
We have a separate [issue](https://github.com/grafana/grafana/issues/29339) for adding UIDs but it makes sense also to change the name to be unique as well. @kylebrandt WDYT?